### PR TITLE
OBPIH-5801 Add approval/rejection fields to the Requisition domain

### PR DESCRIPTION
**Pull request by kchelstowski**

### CODE REVIEW

File: grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy
Comment (kchelstowski) [inline grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy:1712]: it's gonna be removed, it was for testing purposes.

File: src/groovy/org/pih/warehouse/requisition/RequisitionStatus.groovy
Comment (kchelstowski) [inline src/groovy/org/pih/warehouse/requisition/RequisitionStatus.groovy:40]: let me know what `sortOrder` those should have in your opinion - I've just added it at the end for now, to have a discussion about that.

File: grails-app/domain/org/pih/warehouse/requisition/Requisition.groovy
Comment (awalkowiak) [inline grails-app/domain/org/pih/warehouse/requisition/Requisition.groovy:96]: Can you move it down to other related fields and add a comment that these are `approval` related fields

File: src/groovy/org/pih/warehouse/requisition/RequisitionStatus.groovy
Comment (awalkowiak) [reply src/groovy/org/pih/warehouse/requisition/RequisitionStatus.groovy:40]: These new fields should be mixed in the first "section" of statuses. As you can see you added these in `// Removed` section.

**Review by awalkowiak — COMMENTED** [review]
I have not yet reviewed the DB change log, but I already found a few issues (see comments) and you forgot about adding a list of designated approvers to that request.

**Review by awalkowiak — CHANGES_REQUESTED** [review]
See my previous comment

**Review by alannadolny — APPROVED** [review]

File: src/groovy/org/pih/warehouse/core/EventCode.groovy
Comment (jmiranda) [inline src/groovy/org/pih/warehouse/core/EventCode.groovy:28]: I'm wondering if we could come up with a more generic term for this like SUBMITTED. Or maybe we need both SUBMITTED and something like PENDING_APPROVAL.

**Review by jmiranda — CHANGES_REQUESTED** [review]

File: grails-app/domain/org/pih/warehouse/requisition/Requisition.groovy
Comment (kchelstowski) [reply grails-app/domain/org/pih/warehouse/requisition/Requisition.groovy:127]: I don't mind doing that, so I leave the decision for you.
On the other hand, I think this could be misleading, unless we'd override the `dateSubmitted` over and over again.
I imagine a situation, when the request is REJECTED, then I believe someone will have a chance to correct it (?), and then again will send it for an approval (submit).

So the question would be - if we reject the request - should the `dateSubmitted` be cleared out or should it stay, and then, after another submission we'd just override it?

File: src/groovy/org/pih/warehouse/requisition/RequisitionStatus.groovy
Comment (kchelstowski) [reply src/groovy/org/pih/warehouse/requisition/RequisitionStatus.groovy:22]: I've consulted the order with Katarzyna and yeah, we've struggled with positioning those two - we agreed they are "equal", so I couldn't decide whether I should place them above or below the VERIFYING.
I'll change that.

File: src/groovy/org/pih/warehouse/core/EventCode.groovy
Comment (kchelstowski) [reply src/groovy/org/pih/warehouse/core/EventCode.groovy:28]: If it were up to me, I'd distinguish them. Having said `PENDING_APPROVAL`, do you mean that I should change the naming from `WAITING_FOR_APPROVAL` to `PENDING_APPROVAL`, or it was just an open suggestion?

File: grails-app/domain/org/pih/warehouse/requisition/Requisition.groovy
Comment (kchelstowski) [reply grails-app/domain/org/pih/warehouse/requisition/Requisition.groovy:129]: honestly I don't have a better idea. The only one that comes to my mind would be `approvalRequired`, but this is the same, but in reversed order, so I don't know.

File: grails-app/domain/org/pih/warehouse/requisition/Requisition.groovy
Comment (kchelstowski) [reply grails-app/domain/org/pih/warehouse/requisition/Requisition.groovy]: fixed

File: grails-app/domain/org/pih/warehouse/requisition/Requisition.groovy
Comment (kchelstowski) [reply grails-app/domain/org/pih/warehouse/requisition/Requisition.groovy]: After consulting with @awalkowiak , we will leave it as it is for now, not to block the whole feature, as we need to discuss it and we'll eventually include this refactor in this ticket but as "post-factum".

File: grails-app/domain/org/pih/warehouse/requisition/Requisition.groovy
Comment (kchelstowski) [reply grails-app/domain/org/pih/warehouse/requisition/Requisition.groovy]: went for `approvalRequired` (approved by @awalkowiak )

File: grails-app/migrations/0.8.x/changelog-2023-09-13-1300-alter-table-requisition-add-request-approval-improvement.xml
Comment (kchelstowski) [reply grails-app/migrations/0.8.x/changelog-2023-09-13-1300-alter-table-requisition-add-request-approval-improvement.xml:40]: since you probably forgot to reply on that one, I left the "manual" foreign keys names for now not to waste time.

File: grails-app/migrations/0.8.x/changelog-2023-09-13-1300-alter-table-requisition-add-request-approval-improvement.xml
Comment (kchelstowski) [reply grails-app/migrations/0.8.x/changelog-2023-09-13-1300-alter-table-requisition-add-request-approval-improvement.xml]: fixed

File: grails-app/domain/org/pih/warehouse/requisition/Requisition.groovy
Comment (kchelstowski) [reply grails-app/domain/org/pih/warehouse/requisition/Requisition.groovy:125]: Discussed with the guys - we have `dateRequested` already, hence we didn't find an advantage of adding it for now.

File: src/groovy/org/pih/warehouse/requisition/RequisitionStatus.groovy
Comment (kchelstowski) [reply src/groovy/org/pih/warehouse/requisition/RequisitionStatus.groovy]: changed the order

File: src/groovy/org/pih/warehouse/core/EventCode.groovy
Comment (kchelstowski) [reply src/groovy/org/pih/warehouse/core/EventCode.groovy]: Changed to `PENDING_APPROVAL`, but left the translation as "Waiting for approval". Additionally - added translations for all of the EventCodes, since there were not any.

**Review by awalkowiak — APPROVED** [review]

**Pull request merged**
